torchtitan/components/tokenizer.py

```diff
@@ -5,7 +5,13 @@
 # LICENSE file in the root directory of this source tree.
 
 
+import json
+import os
 from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from tokenizers import Tokenizer as HfTokenizer
+from typing_extensions import override
 
 
 class Tokenizer(ABC):
```

Comment:
```suggestion
class BaseTokenizer(ABC):
```
similar to https://github.com/pytorch/torchtitan/blob/main/torchtitan/components/dataloader.py#L26

Reply:
Agree on this, I had originally updated this to `BaseTokenizer`, but it also required a lot of changes to many parts of the code that reference this class including a lot of changes in the `experiments/`. So for better understanding of this PR I split these changes off.

I can add it back in if you like, but otherwise would prefer to have it in a separate refactor PR.